In [0]:
student_scores_high = np.random.randint(low = 80, high=100, size =(30, 3)) 
student_scores_middle = np.random.randint(low = 50, high=79, size =(20, 3)) 
student_scores_low = np.random.randint(low = 0, high=49, size =(100, 3)) 

In [0]:
import numpy as np

# high_score_students

In [0]:
# high
# 국영수
# 80 ~ 100점 까지 30명의 점수

student_scores_high

국어

In [0]:
# 1. 점수가 높은 학생들의 국어 점수의 평균이 점수가 높은 학생들의 대표값(특징)이 될 수 있는지를 확인해보기

# 점수가 높은 학생들의 국어 점수의 평균
np.mean(student_scores_high[:,0])

In [0]:
# min, max
# 평균 86.9 가 최소값 80 에 가깝고 최대값인 97과는 너무 먼 느낌

np.min(student_scores_high[:,0]), np.max(student_scores_high[:,0])

(80, 97)

In [0]:
# var, std
# 산포도 
# 표준편차가 000에 가까울수록 측정값이 평균(점선)에 가까움을 의미합니다. 측정값이 평균으로부터 멀리 떨어져 있을수록 표준편차는 커집니다.
# std(표준편차) 가 0과 멀기 때문에 측정값이 평균과 크게 맞지 않고 많이 퍼져, 넓게 산포되어 있음을 알 수 있다.

np.var(student_scores_high[:,0]), np.std(student_scores_high[:,0])

(24.22333333333334, 4.9217205663602375)

In [0]:
 # median
 # 중앙값인 87.5 랑 평균 86.9 랑 별 차이 없어 보임

 np.median(student_scores_high[:,0])

87.5

영어

In [0]:
# 2. 점수가 높은 학생들의 영어 점수의 평균이 점수가 높은 학생들의 대표값(특징)이 될 수 있는지를 확인해보기

# 점수가 높은 학생들의 영어 점수의 평균
np.mean(student_scores_high[:,1])

90.43333333333334

In [0]:
# min, max
# 평균 90.43 이 최소값 83과 최대값 98의 중간에 적당히 자리한 느낌

np.min(student_scores_high[:,1]), np.max(student_scores_high[:,1])

(83, 98)

In [0]:
# var, std
# 산포도 
# 표준편차가 000에 가까울수록 측정값이 평균(점선)에 가까움을 의미합니다. 측정값이 평균으로부터 멀리 떨어져 있을수록 표준편차는 커집니다.
# std(표준편차) 가 0과 멀기 때문에 측정값이 평균과 크게 맞지 않고 많이 퍼져, 넓게 산포되어 있음을 알 수 있다.

np.var(student_scores_high[:,1]), np.std(student_scores_high[:,1])

(17.312222222222225, 4.160795864041185)

In [0]:
# median
# 중앙값인 90.0 랑 평균 90.43 랑 별 차이 없어 보임

np.median(student_scores_high[:,1])

90.0

수학

In [0]:
# 3. 점수가 높은 학생들의 수학 점수의 평균이 점수가 높은 학생들의 대표값(특징)이 될 수 있는지를 확인해보기

# 점수가 높은 학생들의 수학 점수의 평균
np.mean(student_scores_high[:,2])

91.1

In [0]:
# min, max
# 평균 91.1 이 최소값 80 보다는 최대값 99 에 조금 더 가까운 느낌

np.min(student_scores_high[:,2]), np.max(student_scores_high[:,2])

(80, 99)

In [0]:
# var, std
# 산포도 
# 표준편차가 000에 가까울수록 측정값이 평균(점선)에 가까움을 의미합니다. 측정값이 평균으로부터 멀리 떨어져 있을수록 표준편차는 커집니다.
# std(표준편차) 가 0과 멀기 때문에 측정값이 평균과 크게 맞지 않고 많이 퍼져, 넓게 산포되어 있음을 알 수 있다.

np.var(student_scores_high[:,2]), np.std(student_scores_high[:,2])

(38.55666666666668, 6.209401474108972)

In [0]:
# median
# 중앙값인 91.0 랑 평균 91.1 랑 별 차이 없음

np.median(student_scores_high[:,2])

91.0

# 점수가 높은 학생들을 대표할 수 있는 값은 무엇인가

국,영,수 의 표준편차를 비교했을때 영 < 국 < 수 로 확인이 가능.\
표준편차가 작을수록 평균에 가깝게 분포했음을 알 수 있음.\
영어와 국어의 표준편차가 각각 4.16, 4.92 이고\
수학의 표준편차가 6.20 이므로\
사실 평균에서 산포는 넓게 되어 군집이 밀집되어 있다고는 보기 힘드나\
세 과목 중에 비교적 영어와 국어의 성적이\
이 점수가 높은 학생들의 특징을 나타낸다고 할 수 있다.

# middle_score_students

In [0]:
# middle
# 국영수
# 50 ~ 79점 까지 20명의 점수

student_scores_middle

국어

In [0]:
# 1. 점수가 중간인 학생들의 국어 점수의 평균이 점수가 중간인 학생들의 대표값(특징)이 될 수 있는지를 확인해보기

# 점수가 중간인 학생들의 국어 점수의 평균
# 63.9
np.mean(student_scores_middle[:,0])

# min, max
# 50, 78
# 평균 63.9 가 최소값 50 최대값 78의 사이에 적당히 자리한 느낌

np.min(student_scores_middle[:,0]), np.max(student_scores_middle[:,0])


# var, std
# 70.39, 8.389874850079709
# 산포도 
# 표준편차가 000에 가까울수록 측정값이 평균(점선)에 가까움을 의미합니다. 측정값이 평균으로부터 멀리 떨어져 있을수록 표준편차는 커집니다.
# std(표준편차) 가 0과 멀기 때문에 측정값이 평균과 크게 맞지 않고 많이 퍼져, 넓게 산포되어 있음을 알 수 있다.

np.var(student_scores_middle[:,0]), np.std(student_scores_middle[:,0])

 # median
 # 63.0
 # 평균이랑 별 차이 없어 보임

 np.median(student_scores_middle[:,0])

In [0]:
np.mean(student_scores_middle[:,0]), np.min(student_scores_middle[:,0]), np.max(student_scores_middle[:,0]), np.var(student_scores_middle[:,0]), np.std(student_scores_middle[:,0]), np.median(student_scores_middle[:,0])

(63.9, 50, 78, 70.39, 8.389874850079709, 63.0)

영어

In [0]:
# 2. 점수가 중간인 학생들의 영어 점수의 평균이 점수가 중간인 학생들의 대표값(특징)이 될 수 있는지를 확인해보기

# 영어 점수의 평균
# 62.15
np.mean(student_scores_middle[:,1])

# min, max
# 50, 78
# 평균이 살짝 최소값과 가까운 느낌

np.min(student_scores_middle[:,1]), np.max(student_scores_middle[:,1])

# var, std
# 95.72749999999999, 9.78404313154843
# 산포도 
# 표준편차가 000에 가까울수록 측정값이 평균(점선)에 가까움을 의미합니다. 측정값이 평균으로부터 멀리 떨어져 있을수록 표준편차는 커집니다.
# std(표준편차) 가 0과 멀기 때문에 측정값이 평균과 크게 맞지 않고 많이 퍼져, 넓게 산포되어 있음을 알 수 있다.

np.var(student_scores_middle[:,1]), np.std(student_scores_middle[:,1])

# median
# 58.5
# 최소값에 매우 가까움
# 점수가 중간인 학생의 영어점수의 절반이 50 ~ 58.5 사이에 위치함.
# 62.15 인 평균보다 58.5 인 중앙값이 영어점수를 대표한다고 말할 수 있음.

np.median(student_scores_middle[:,1])

In [0]:
np.mean(student_scores_middle[:,1]), np.min(student_scores_middle[:,1]), np.max(student_scores_middle[:,1]), np.var(student_scores_middle[:,1]), np.std(student_scores_middle[:,1]), np.median(student_scores_middle[:,1])

(62.15, 50, 78, 95.72749999999999, 9.78404313154843, 58.5)

수학

In [0]:
# 3. 점수가 중간인 학생들의 수학 점수의 평균이 점수가 중간인 학생들의 대표값(특징)이 될 수 있는지를 확인해보기

# 수학 점수의 평균
# 65.75
np.mean(student_scores_middle[:,2])

# min, max
# 52, 78
# 평균 이 최소값에 조금 더 가까운 느낌

np.min(student_scores_middle[:,2]), np.max(student_scores_middle[:,2])

# var, std
# 64.4875, 8.030410948388631
# 산포도 
# 표준편차가 000에 가까울수록 측정값이 평균(점선)에 가까움을 의미합니다. 측정값이 평균으로부터 멀리 떨어져 있을수록 표준편차는 커집니다.
# std(표준편차) 가 0과 멀기 때문에 측정값이 평균과 크게 맞지 않고 많이 퍼져, 넓게 산포되어 있음을 알 수 있다.

np.var(student_scores_middle[:,2]), np.std(student_scores_middle[:,2])

# median
# 65.0
# 중앙값과 평균이 별 차이 없음

np.median(student_scores_middle[:,2])

In [0]:
np.mean(student_scores_middle[:,2]), np.min(student_scores_middle[:,2]), np.max(student_scores_middle[:,2]), np.var(student_scores_middle[:,2]), np.std(student_scores_middle[:,2]), np.median(student_scores_middle[:,2])

(65.75, 52, 78, 64.4875, 8.030410948388631, 65.0)

#점수가 중간인 학생들을 대표할 수 있는 값은 무엇인가

국영수 표준편차는 각각 8.38, 9.78, 8.03\
수 < 국 < 영 순서대로 군집이 더 넓게 퍼져있음.\
점수가 중간인 학생들을 대표하는 특징은 수학과 국어라고 할 수 있음.\
특히 영어는 평균보다는 중앙값이 더 대표값에 적합함.

# low_score_students

In [0]:
# low
# 국영수
# 0 ~ 49점 까지 100명의 점수

student_scores_low

국어

In [0]:
# 1. 점수가 낮은 학생들의 국어 점수의 평균이 점수가 낮은 학생들의 대표값(특징)이 될 수 있는지를 확인해보기

# 국어 점수의 평균
# 24.01
np.mean(student_scores_low[:,0])

# min, max
# 0, 48
# 평균 24.01이 최소값에 조금 더 가까운 느낌. 

np.min(student_scores_low[:,0]), np.max(student_scores_low[:,0])


# var, std
# 235.6299, 15.350241040452753
# 산포도 
# 표준편차가 000에 가까울수록 측정값이 평균(점선)에 가까움을 의미합니다. 측정값이 평균으로부터 멀리 떨어져 있을수록 표준편차는 커집니다.
# std(표준편차) 가 0과 멀기 때문에 측정값이 평균과 크게 맞지 않고 많이 퍼져, 넓게 산포되어 있음을 알 수 있다.

np.var(student_scores_low[:,0]), np.std(student_scores_low[:,0])

 # median
 # 22.5
 # 중앙값이랑 평균이랑 별 차이 없어 보임

 np.median(student_scores_low[:,0])

In [0]:
np.mean(student_scores_low[:,0]), np.min(student_scores_low[:,0]), np.max(student_scores_low[:,0]), np.var(student_scores_low[:,0]), np.std(student_scores_low[:,0]), np.median(student_scores_low[:,0])

(24.01, 0, 48, 235.6299, 15.350241040452753, 22.5)

영어

In [0]:
# 2. 점수가 낮은 학생들의 국어 점수의 평균이 점수가 낮은 학생들의 대표값(특징)이 될 수 있는지를 확인해보기

# 영어 점수의 평균
# 23.72
np.mean(student_scores_low[:,1])

# min, max
# 0, 47
# 평균이 최소값과 최대값 중간에 적당히 자리한 느낌

np.min(student_scores_low[:,1]), np.max(student_scores_low[:,1])

# var, std
# 208.5416, 14.440969496540044
# 산포도 
# 표준편차가 000에 가까울수록 측정값이 평균(점선)에 가까움을 의미합니다. 측정값이 평균으로부터 멀리 떨어져 있을수록 표준편차는 커집니다.
# std(표준편차) 가 0과 멀기 때문에 측정값이 평균과 크게 맞지 않고 많이 퍼져, 넓게 산포되어 있음을 알 수 있다.

np.var(student_scores_low[:,1]), np.std(student_scores_low[:,1])

# median
# 23.0
# 평균이랑 거의 차이 없음

np.median(student_scores_low[:,1])

In [0]:
np.mean(student_scores_low[:,1]), np.min(student_scores_low[:,1]), np.max(student_scores_low[:,1]), np.var(student_scores_low[:,1]), np.std(student_scores_low[:,1]), np.median(student_scores_low[:,1])

(23.72, 0, 47, 208.5416, 14.440969496540044, 23.0)

수학

In [0]:
# 3. 점수가 낮은 학생들의 국어 점수의 평균이 점수가 낮은 학생들의 대표값(특징)이 될 수 있는지를 확인해보기

# 수학 점수의 평균
# 23.96
np.mean(student_scores_low[:,2])

# min, max
# 0, 48
# 평균이 최소값과 최대값 사이에 적당히 자리한 느낌
np.min(student_scores_low[:,2]), np.max(student_scores_low[:,2])

# var, std
# 179.11839999999995, 13.383512244549259
# 산포도 
# 표준편차가 000에 가까울수록 측정값이 평균(점선)에 가까움을 의미합니다. 측정값이 평균으로부터 멀리 떨어져 있을수록 표준편차는 커집니다.
# std(표준편차) 가 0과 멀기 때문에 측정값이 평균과 크게 맞지 않고 많이 퍼져, 넓게 산포되어 있음을 알 수 있다.

np.var(student_scores_low[:,2]), np.std(student_scores_low[:,2])

# median
# 23.5
# 중앙값이 평균이랑 별 차이 없음

np.median(student_scores_low[:,2])

In [0]:
np.mean(student_scores_low[:,2]), np.min(student_scores_low[:,2]), np.max(student_scores_low[:,2]), np.var(student_scores_low[:,2]), np.std(student_scores_low[:,2]), np.median(student_scores_low[:,2])

(23.96, 0, 48, 179.11839999999995, 13.383512244549259, 23.5)

#점수가 낮은 학생들을 대표할 수 있는 값은 무엇인가

국영수 표준편차는 각각 15.35,14.44,13.38\
수 < 영 < 국 순서대로 표준편차의 크기가 큼\
수학의 평균이 점수가 낮은 학생들을 대표한다고 할 수 있으나\
세 과목 모두 분산과 표준편차가 아주 크므로\
산포가 아주 넓게 퍼져있어\
특정하게 어떤 과목이 이 점수가 낮은 학생들을 대표한다고 말하기 어렵다.
